generator/src/main/java/com/scottlogic/deg/generator/generation/DecisionTreeDataGenerator.java

```diff
@@ -46,7 +59,7 @@ public DecisionTreeDataGenerator(
 
         final Stream<Stream<RowSpec>> rowSpecsByPartition = partitionedTrees
             .stream()
-            .map(treeWalker::walk);
+            .map(treeWalkerFactory.getDecisionTreeWalker(decisionTree)::walk);
```

Comment:
Do you expect the walker to vary per partition, if so this is fine. If not the call to get the walker should be prior to the stream creation (using the original tree as the input).
Currently this will (potentially) select/create a new walker for each partition.